sympy/assumptions/handlers/sets.py

```diff
@@ -336,6 +338,12 @@ class AskHermitianHandler(AskRealHandler):
     Test that an expression belongs to the field of Hermitian operators
     """
 
+    @staticmethod
+    def Expr(expr, assumptions):
+        if isinstance(expr, MatrixBase):
```

Comment:
subclasses should have their own method. Define `def MatrixBase( ... ):` instead of using `isinstance`.

Reply:
Yes, actually, according to the current logic, if subclasses, give `None` then the call is transferred to the methods up in the hierarchy in the `__mro__`. So, probably, a fix is needed somewhere there.